### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Ntwali-Josue**
### In this Pull Request, I:

- [x] Add `Corrector` class
- [x] Add `validate_name` method in `Person` class

### CODE REVIEW

File: corrector.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Corrector
+  def initialize(name)
+    @name = name
+  end
```
Comment (TedLivist) [inline corrector.rb:4]: - [ ] The `initialize` method is not really necessary. In keeping with the YAGNI rule, I would suggest you remove it. It contains the `@name` instance variable but as you can see you did not really use it at all. The `correct_name` method should just be enough for this class.

File: corrector.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+  def correct_name(name)
+    name.capitalize
+    name.slice(0, 10) if name.length > 10
+  end
```
Comment (TedLivist) [inline corrector.rb:9]: - [ ] Your `correct_name` method does not work as expected. It does not capitalize the `name` parameter (see image). You can resolve this by adding the bang (`!`) sign to capitalize method like so: `capitalize!`

  nwtali-corr

- [ ] The `if` conditional in the `correct_name` method is not necessary. Kindly remove it. The `slice` method returns 10 characters anyways without the conditional.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Ntwali-Josue,

Good job so far!
✔️ Good documentation
✔️ There is a Corrector class
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
#### _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @Ntwali-Josue, 

Good job on implementing the suggested changes. Well Done.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**